### PR TITLE
Fix deep_iterable, deep_mapping validator type stubs

### LINKED ISSUE #531: Typing issue with deep_iterable validator
When I run mypy check on the following code:

```python

import attr
from attr.validators import deep_iterable, instance_of
from typing import List


@attr.s
class Example(object):
    x = attr.ib(
        type=List[int],
        validator=deep_iterable(instance_of(int), instance_of(list))
    )
```
it results in:

```
script.py:11: error: Cannot infer type argument 1 of "deep_iterable"
script.py:11: error: Argument 1 to "instance_of" has incompatible type "Type[int]"; expected "Union[Tuple[Type[List[int]], ...], Type[List[int]]]"
script.py:11: error: Argument 1 to "instance_of" has incompatible type "Type[List[Any]]"; expected "Union[Tuple[Type[List[int]], ...], Type[List[int]]]"
```
mypy == 0.670, attrs == 19.1.0

### PR DESCRIPTION
**Pull request by mmaslowskicc**
Use bound type variables for iterables and mappings of the validated item types, so mypy does not erroneously require
both the iterable attribute validated by deep_iterable and the item type validated by the first argument of
deep_iterable to have the same type. This fixes #531.

Do not require optional arguments in type stubs.

# Pull Request Check List

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](https://www.attrs.org/en/latest/contributing.html) at least once, it will save you unnecessary review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] Added **tests** for changed code.
- [x] New features have been added to our [Hypothesis testing strategy](https://github.com/python-attrs/attrs/blob/master/tests/strategies.py).
- [x] Changes or additions to public APIs are reflected in our type stubs (files ending in ``.pyi``).
    - [x] ...and used in the stub test file `tests/typing_example.py`.
- [x] Updated **documentation** for changed code.
    - [x] New functions/classes have to be added to `docs/api.rst` by hand.
    - [x] Changes to the signature of `@attr.s()` have to be added by hand too.
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded).
- [x] Documentation in `.rst` files is written using [semantic newlines](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).
- [ ] Changes (and possible deprecations) have news fragments in [`changelog.d`](https://github.com/python-attrs/attrs/blob/master/changelog.d).

If you have *any* questions to *any* of the points above, just **submit and ask**!  This checklist is here to *help* you, not to deter you from contributing!

### CODE REVIEW

**Comment by euresti** [comment]
The stub changes make sense to me. Thanks for doing them!

**Comment by hynek** [comment]
Can you review and merge please? I’m traveling and I was gonna ping you anyways. :)

File: src/attr/validators.pyi [hunk 1]
```diff
@@ -14,11 +29,11 @@ def in_(options: Container[_T]) -> _ValidatorType[_T]: ...
 def and_(*validators: _ValidatorType[_T]) -> _ValidatorType[_T]: ...
 def deep_iterable(
     member_validator: _ValidatorType[_T],
-    iterable_validator: Optional[_ValidatorType[_T]],
-) -> _ValidatorType[_T]: ...
+    iterable_validator: Optional[_ValidatorType[_I]] = None,
```
Comment (euresti) [inline src/attr/validators.pyi]: I know the default value is actually `None` but in stubs you usually just write `= ...`

File: src/attr/validators.pyi [hunk 2]
```diff
@@ -14,11 +29,11 @@ def in_(options: Container[_T]) -> _ValidatorType[_T]: ...
...
+) -> _ValidatorType[_I]: ...
 def deep_mapping(
-    key_validator: _ValidatorType[_T],
-    value_validator: _ValidatorType[_T],
-    mapping_validator: Optional[_ValidatorType[_T]],
-) -> _ValidatorType[_T]: ...
+    key_validator: _ValidatorType[_K],
+    value_validator: _ValidatorType[_V],
+    mapping_validator: Optional[_ValidatorType[_M]] = None,
```
Comment (euresti) [inline src/attr/validators.pyi]: Same here

**Comment by euresti** [comment]
Thanks for the fix!

**Pull request merged**